src/senaite/app/listing/adapters/workflow.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+
+from bika.lims import api
+from senaite.app.listing import logger
+from senaite.app.listing import senaiteMessageFactory as _
+from senaite.app.listing.interfaces import IListingWorkflowTransition
+from ZODB.POSException import ConflictError
+from zope.interface import implementer
+
+
+@implementer(IListingWorkflowTransition)
+class ListingWorkflowTransition(object):
+    """Adapter to execute workflow transitions in listings
+    """
+    def __init__(self, view, context, request):
+        self.view = view
+        self.context = context
+        self.request = request
+        self.error = {}
+
+    @property
+    def failed(self):
+        """Flag to indicate if the transition failed
+        """
+        if self.error:
+            return True
+        return False
+
+    def get_error(self):
+        """Return the error message
+        """
+        return self.error.get("message")
+
+    def get_redirect_url(self):
+        """Return redirect URL
+        """
+        return ""
+
+    def do_transition(self, transition):
+        """Execute the workflow transition
+        """
+        obj = self.context
+        oid = api.get_id(obj)
+
+        try:
+            obj = api.do_transition_for(obj, transition)
+        except ConflictError:
+            self.error["message"] = _(
+                "A database conflict occured during transition "
```

Comment:
typo: *occured* --> *occurred*

I would also rephrase as *A database conflict error occurred during ...*

Reply:
Fixed in 11af065